src/main/resources/templates/add.html

```diff
@@ -0,0 +1,76 @@
+<!DOCTYPE html>
+<html lang="en" xmlns:th="http://www.thymeleaf.org">
+<head th:include="fragments :: head">
+</head>
+<body>
+<div th:replace="fragments :: header"></div>
+<main>
+  <div class="container text-center form "  >
+    <form th:action="@{/save}" method="post" th:object="${recipeToCreate}">
+      <div class="mb-3">
+        <input th:field="*{id}" hidden="">
+        <div class="mb-3">
+          <label for="difficultyLevel" class="form-label">Poziom trudności</label>
+          <select class="form-select" id="difficultyLevel" th:field="*{difficultyLevel}">
```

Comment:
Z tego co widzę to klasa `form-select` jest z Bootstrapa 5, a w importach widziałem wersje 4.x.
Na razie ten select nie jest ostylowany. Możesz albo zastosować klasy z Boostrapa 4, albo zaimportowac boostrapa 5.